src/fs.toit

```diff
@@ -3,75 +3,312 @@
 // found in the package's LICENSE file.
 
 import system
+import .posix as posix
+import .windows as windows
+
+/**
+A library to work with paths.
+
+All functions in this library work syntactically on paths. They do not access
+  the file system.
+
+# Windows
+Windows introduces lots of complications compared to Posix systems. This
+  section describes some of the concepts.
+
+## Path separators
+Windows allows both forward slashes and backslashes as path separators.
+  This library keeps the original separators when possible, but always
+  uses backslashes when introducing new separators. The $clean function
+  furthermore replaces all forward slashes with backslashes.
+
+## Types of paths
+Drive paths are paths that start with a drive letter and a colon. For
+  example `c:`.
+
+Drive absolute paths are drive paths that are fully qualified (absolute).
+  For example `c:\\path\\to\\file`. They start with a drive letter and a
+  colon, followed by a path that is rooted on the drive.
+
+Drive relative paths are drive paths that are relative to the current
+  working directory of the drive. For example, `c:foo\\bar`. They start
+  with a drive letter and a colon, followed by a path that is *not*
+  rooted on the drive. This type of path is a relict from the DOS days
+  and should not be used anymore. On current versions of Windows there
+  isn't a drive-specific working directory anymore, although there are hidden
+  environment variables to indicate drive-specific current working
+  directories. If the drive of a drive-relative path is the same
```

Comment:
Drive relative or drive-relative. I'd probably pick the former because it works better for the absolute variant.

Reply:
done.